drivers/samsung/displays/mdc_protocol.cr

```diff
@@ -0,0 +1,364 @@
+    Hdmi2         = 0x23
+    Hdmi2_Pc      = 0x24
+    Hdmi3         = 0x31
+    Hdmi3_Pc      = 0x32
+    Hdmi4         = 0x33
+    Hdmi4_Pc      = 0x34
+    Display_Port  = 0x25
+    Dtv           = 0x40
+    Media         = 0x60
+    Widi          = 0x61
+    Magic_Info    = 0x20
+    Whiteboard    = 0x64
+  end
+  include PlaceOS::Driver::Interface::InputSelection(Inputs)
+
+  # Discovery Information
+  tcp_port 1515
+  descriptive_name "Samsung MD, DM & QM Series LCD"
+  generic_name :Display
+
+  # Markdown description
+  description <<-DESC
+  For DM displays configure the following 1:
+
+  1. Network Standby = ON
+  2. Set Auto Standby = OFF
+  3. Set Eco Solution, Auto Off = OFF
+
+  Hard Power off displays each night and hard power ON in the morning.
+  DESC
+
+  default_settings({
+    display_id: 0,
+    rs232_control: false
+  })
+
+  @id : Int32 = 0
+  @rs232 : Bool = false
+  @blank : Input?
+  @previous_volume : Int32 = 50
+  @input_stable : Bool = true
+  @input_target : Input? = nil
+  @power_stable : Bool = true
+
+  def on_load
+    transport.tokenizer = Tokenizer.new do |io|
+      bytes = io.peek
+      # Ensure message indicator is well-formed
+      disconnect unless bytes.first == INDICATOR
+      logger.debug { "Received: #{bytes}" }
+      # [header, command, id, data.size, [data], checksum]
+      # return 0 if the message is incomplete
+      bytes.size < 4 ? 0 : bytes[3].to_i + 5
+    end
+
+    on_update
+  end
+
+  def on_update
+    @id = setting(Int32, :display_id)
+    @rs232 = setting(Bool, :rs232_control)
+    if blanking_input = setting?(String, :blanking_input)
+      @blank = Input.parse?(blanking_input)
+    end
+  end
+
+  def connected
+    do_device_config unless (hard_off = self[:hard_off]?) && hard_off.as_bool?
+
+    schedule.every(30.seconds, true) do
+      do_poll
+    end
+  end
+
+  def disconnected
+    self[:power] = false unless @rs232
+    schedule.clear
+  end
+
+  # As true power off disconnects the server we only want to power off the panel
+  def power(power : Bool)
+    self[:power_target] = power
+    @power_stable = false
+
+    if !power
+      # Blank the screen before turning off panel if required
+      # required by some video walls where screens are chained
+      if (blanking_input = @blank) && self[:power]?
+        switch_to(blanking_input)
+      end
+      do_send(Command::Panel_Mute, 1)
+    else
+      # Power on
+      do_send(Command::Hard_Off, 1)
+      do_send(Command::Panel_Mute, 0)
+    end
+  end
+
+  def hard_off
+    do_send(Command::Panel_Mute, 0) if (power = self[:power]?) && power.as_bool?
+    do_send(Command::Hard_Off, 0)
+  end
+
+  def power?(**options)
+    do_send(Command::Panel_Mute, Bytes.empty, **options).get
+    self[:power]
+  end
+
+  # Mutes both audio/video
+  def mute(
+    state : Bool = true,
+    index : Int32 | String = 0,
+    layer : MuteLayer = MuteLayer::AudioVideo
+  )
+    mute_video(state) unless layer == MuteLayer::Audio
```

Comment:
Rather than using an negative condition, this is better suited to matching a specific layer (either an `if` or a `case` statement). This way if other items are added to the MuteLayer enum, it won't impact behaviour here.